src/aero_state.F90

```diff
@@ -820,13 +869,18 @@ subroutine aero_state_sample_particles(aero_state_from, aero_state_to, &
     integer, intent(in) :: removal_action
 
     integer :: n_transfer, i_transfer, i_part
+    integer :: i_group, i_class
     logical :: do_add, do_remove
     real(kind=dp) :: num_conc_from, num_conc_to
     type(aero_info_t) :: aero_info
+    integer :: n_parts_before
 
     call assert(721006962, (sample_prob >= 0d0) .and. (sample_prob <= 1d0))
+#ifndef PMC_USE_WRF
```

Comment:
@mwest1066 This is something a bit weird that I ran across in WRF-PartMC where I need to not do the following. However the comment above for this subroutine, says that the weight of `aero_state_to` should already be set. However the code always copies the weight  in the box model.

Reply:
This is only called in one place (in `scenario_update_aero_state()`). I think we should change that caller to do the zero/copy_weight and remove them from this function. That would make the comment correct and also remove the need for the `PMC_USE_WRF` check.

Reply:
Fixed in https://github.com/compdyn/partmc/pull/168/commits/522b269df1f3ed384a2c6047cd0660bd0abe6c5b